script.js

```diff
@@ -0,0 +1,200 @@
+    });
+    const filteredBrand = Promise.all(finalResult)
+    .then((res)=>{
+        res.map((brand,index, array) => {
+          if(brand.length > 0){
+          linksList.forEach(function(product) {
+            const node = product.childNodes[0];
+            const label = node.getAttribute("aria-label");
+            let productNodeText = label.toLowerCase();
+            if(productNodeText.includes(brand[0].toLowerCase())){
+                product.style["border-color"] =resultbrands.colourMap[resultbrands.result[index].OverallScore];
+                product.classList.add('product');
+                var button = document.createElement("button");
+                button.innerText = "More information";
+                button.setAttribute("id", "btn");
+                product.appendChild(button);
+
+                var span =document.createElement("span");
+                var myModal =document.createElement("div");
+
+                if(button){
+                button.onclick = function() {
+                  myModal.classList.add('myModal');
+                  myModal.setAttribute("id", "myModal");
+                  var content =document.createElement("div");
+                  content.classList.add('popup-content');
+                  var header =document.createElement("div");
+                  header.classList.add('popup-header');
+                  span.classList.add('end');
+                  span.innerText="X";
+                  var score=document.createElement('h1');
+                  score.innerText="Brand Ratings";
+                  score.classList.add('title');
+
+                  //LaborScore
+                  var LS=document.createElement('div');
+                  LS.classList.add('ls');
+
+                  var LaborScore=document.createElement('span');
+                  LaborScore.classList.add('score');
+
+                  if(resultbrands.result[index].LaborScore){
+                  LaborScore.innerText=resultbrands.result[index].LaborScore;
+                  LaborScore.style["background-color"] =resultbrands.colourMap[resultbrands.result[index].LaborScore];
+                }
+                else{
+                  LaborScore.innerText="TBD";
+                  LaborScore.style["background-color"] =resultbrands.colourMap[resultbrands.result[index].LaborScore];
+                }
+                  var LaborScoret=document.createElement('span');
+                  LaborScoret.classList.add('scoret');
+                  LaborScoret.innerText=" Labor Rating "
+
+                  //TransparencyScore
+                  var TS=document.createElement('div');
+                  TS.classList.add('ls');
+
+                  var TransparencyScore=document.createElement('span');
+                  TransparencyScore.classList.add('score');
+
+                  if(resultbrands.result[index].TransparencyScore){
+                  TransparencyScore.innerText=resultbrands.result[index].TransparencyScore;
```

Comment:
Maybe abstract this long variable in a shorter one, so instead of referencing this long variable each time you need to use it, just give it a new name and use it to reference it, like this:

```js
 let currentTransperancyScore = resultbrands.result[index].TransparencyScore

 TransparencyScore.innerText= currentTransperancyScore;
```